var/spack/repos/builtin/packages/slepc/package.py

```diff
@@ -79,13 +99,18 @@ def install(self, spec, prefix):
             options.extend([
                 '--with-arpack-dir=%s' % spec['arpack-ng'].prefix.lib,
             ])
+            if spec.satisfies('@:3.12.99'):
+                arpackopt='--with-arpack-flags'
+            else:
+                arpackopt='--with-arpack-lib'
+
             if 'arpack-ng~mpi' in spec:
                 options.extend([
-                    '--with-arpack-flags=-larpack'
+                    arpackopt+'=-larpack'
```

Comment:
```suggestion
                    arpackopt + '=-larpack'
```
Also, you can replace `extend([...])` with `append(...)`